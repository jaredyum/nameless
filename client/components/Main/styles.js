export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%',
    [theme.breakpoints.up('md')]: {
      maxWidth: 'calc(100% - 240px)'
    }
  },
  toolbar: theme.mixins.toolbar
});
