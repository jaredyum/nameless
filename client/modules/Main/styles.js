export default theme => ({
  main: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column'
  },
  toolbar: theme.mixins.toolbar
});
