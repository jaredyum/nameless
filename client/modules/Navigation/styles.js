/**
 * The default width of the drawer (sidebar).
 * @const {number}
 */
const DRAWER_WIDTH = 240;

/**
 * The parent styles for this modules.
 * @param {!Object} theme The MUI theme object.
 * @return {!Object.<string, !Object>} The generated styles.
 */
export default theme => ({
  active: {
    fontWeight: 'bold'
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  drawerPaper: {
    width: DRAWER_WIDTH,
    [theme.breakpoints.up('md')]: {
      position: 'relative'
    }
  },
  flex: {
    flex: 1
  },
  link: {
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none',
    padding: '5px 0'
  },
  menuItem: {
    height: 'auto',
    padding: 0,
    color: '#333',
    cursor: 'pointer',
    width: '100%',
    textDecoration: 'none'
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  },
  toolbar: {
    backgroundColor: theme.palette.primary.main,
    color: '#fff',
    ...theme.mixins.toolbar
  }
});
