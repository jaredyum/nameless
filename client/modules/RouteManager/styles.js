export default {
  routeManager: {
    flex: 1,
    display: 'flex'
  }
};
