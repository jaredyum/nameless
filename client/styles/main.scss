// Vendors
@import './vendors/normalize';
@import './vendors/reset';

// Modules
@import './modules/defaults';
