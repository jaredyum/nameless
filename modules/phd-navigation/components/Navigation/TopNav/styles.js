export default theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none'
    }
  },
  flex: {
    flex: 1
  },
  searchBox: {
    borderRadius: '5px',
    padding: '4px 0 2px 10px',
    border: '1px solid #ccc',
    textAlign: 'left'
  }
});
