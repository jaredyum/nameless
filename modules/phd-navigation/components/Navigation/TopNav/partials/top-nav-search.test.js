import {
  getComponentWrapper,
  testCommonComponentAttrs
} from 'utils/tests';

// Styles
import styles from '../styles';

// Components
import TopNavSearch, { DEFAULT_SEARCH_PLACEHOLDER } from './top-nav-search';

const defaultProps = {
  classes: styles
};

const mockEvent = {
  target: { value: 'foobar' },
  preventDefault: jest.fn()
};

let wrapper;
let wrapperInstance;
let searchHandler;

describe('<TopNavSearch /> Component', () => {
  testCommonComponentAttrs(TopNavSearch, defaultProps);

  beforeEach(() => {
    searchHandler = jest.fn();
    defaultProps.searchHandler = searchHandler;
    wrapper = getComponentWrapper(TopNavSearch, defaultProps);
    wrapperInstance = wrapper.instance();
  });

  describe('basic rendering', () => {
    it('should render with a form and input', () => {
      expect(wrapper.find('form')).toHaveLength(1);
      expect(wrapper.find('input')).toHaveLength(1);
    });

    it('should default the placeholder text', () => {
      const input = wrapper.find('input');
      const { placeholder } = input.props();
      expect(placeholder).toEqual(DEFAULT_SEARCH_PLACEHOLDER);
    });
  });

  describe('instance methods', () => {
    it('should set the search value to an empty string', () => {
      const { searchValue } = wrapper.state();
      expect(searchValue).toEqual('');
    });

    it('should update the search value', () => {
      // Trigger a search update.
      wrapperInstance.handleSearchChange(mockEvent);
      // Assert the value was updated.
      const { searchValue } = wrapper.state();
      expect(searchValue).toEqual(mockEvent.target.value);
    });

    it('fire the provided search handler', () => {
      // Trigger a search update, and then submit.
      wrapperInstance.handleSearchChange(mockEvent);
      wrapperInstance.handleSearchSubmit(mockEvent);
      // Assert the handler was called with the proper values.
      expect(searchHandler).toHaveBeenCalledWith(mockEvent.target.value);
    });
  });
});
