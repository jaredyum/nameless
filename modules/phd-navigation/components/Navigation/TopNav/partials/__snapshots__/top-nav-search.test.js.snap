// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavSearch /> Component when rendering should render a component matching the snapshot 1`] = `
<form
  autoComplete="off"
  onSubmit={[Function]}
>
  <input
    id="name"
    label="Name"
    onChange={[Function]}
    placeholder="Search"
    value=""
  />
</form>
`;
